include/aikido/planner/vectorfield/VectorFieldPlanner.hpp

```diff
@@ -2,58 +2,88 @@
 #define AIKIDO_PLANNER_VECTORFIELD_VECTORFIELDPLANNER_HPP_
 
 #include <functional>
+#include <dart/common/Timer.hpp>
 #include <aikido/constraint/Testable.hpp>
+#include <aikido/planner/vectorfield/ConfigurationSpaceVectorField.hpp>
+#include <aikido/planner/vectorfield/VectorFieldPlannerExceptions.hpp>
+#include <aikido/planner/vectorfield/VectorFieldPlannerStatus.hpp>
+#include <aikido/planner/vectorfield/VectorFieldUtil.hpp>
 #include <aikido/statespace/dart/MetaSkeletonStateSpace.hpp>
 #include <aikido/trajectory/Spline.hpp>
 
 namespace aikido {
 namespace planner {
 namespace vectorfield {
 
-enum class VectorFieldPlannerStatus
+/// VectorField Planner generates a trajectory by following a vector field
+/// defined in joint space.
+/// A planned trajectroy ends either by a predefined termination criterion or a
+/// integral time.
+///
+/// This class defines two callback functions for a integrator.
+/// step() provides joint velocities the vector field planner should follow,
+/// check() is called after each integration step to check planner status.
+class VectorFieldPlanner
 {
-  TERMINATE,
-  CACHE_AND_TERMINATE,
-  CACHE_AND_CONTINUE,
-  CONTINUE
-};
+public:
+  /// Constructor
+  ///
+  /// \param[in] _vectorField Vector field in configuration space
+  /// \param[in] _constraint Constraint to be satisfied
 
-/// Callback function of joint velocity calculated by a vector field and a
-/// MetaSkeleton.
-///
-/// \param[in] _stateSpace MetaSkeleton state space
-/// \param[in] _t Planned time of a given duration
-/// \param[out] _dq Joint velocity calculated by a vector field and meta
-/// skeleton
-/// \return Whether vectorfield evaluation succeeds
-using VectorFieldCallback = std::function<bool(
-    const aikido::statespace::dart::MetaSkeletonStateSpacePtr& _stateSpace,
-    double _t,
-    Eigen::VectorXd& _dq)>;
+  VectorFieldPlanner(
+      const aikido::planner::vectorfield::ConfigurationSpaceVectorFieldPtr
```

Comment:
Nit: We don't need to specify the whole namespaces. This can be simplified as
```c++
const ConfigurationSpaceVectorFieldPtr
```

Please remove `aikido::[planner::[vectorfield]]` in this header.